External/Plugins/FlashDebugger/Controls/StackframeUI.cs

```diff
@@ -40,81 +166,346 @@ public StackframeUI(PluginMain pluginMain, ImageList imageList)
+                for (int i = 0, count = lv.Items.Count; i < count; i++)
                 {
-                    item.ImageIndex = -1;
-                    break;
+                    var itemData = (ListItemData)lv.Items[i].Tag;
+                    if (itemData.Index < index) continue;
+                    if (itemData.Index == index)
+                    {
+                        lv.Items[i].ImageIndex = currentImageIndex;
+                        lastSelected = i;
+                        break;
+                    }
+                    if (itemData.Index > index)
+                    {
+                        lv.Items[i - 1].ImageIndex = currentImageIndex;
+                        lastSelected = i - 1;
+                        break;
+                    }
+                    if (i == count - 1 && itemData.Index == -1)
+                    {
+                        lv.Items[i].ImageIndex = currentImageIndex;
+                        lastSelected = i;
+                        break;
+                    }
                 }
             }
-			lv.SelectedItems[0].ImageIndex = currentImageIndex;
-		}
+            else
+            {
+                lastSelected = PluginMain.debugManager.CurrentFrame;
+                selectedItem.ImageIndex = currentImageIndex;
+            }
+        }
 
         public void AddFrames(Frame[] frames)
         {
-            lv.Items.Clear();
+            wholeFrameStack.Clear();
             if (frames.GetLength(0) > 0)
             {
+                Project project = PluginBase.CurrentProject as Project;
+                int i = 0;
                 foreach (Frame item in frames)
                 {
-					String title = item.getCallSignature();
-                    if (item.getLocation().getFile() != null) title += " at " + item.getLocation().getFile() + ":" + item.getLocation().getLine();
-                    lv.Items.Add(new ListViewItem(new string[] {"", title}, -1));
+                    String title = item.getCallSignature();
+                    bool ownFile = false;
+                    SourceFile sourceFile = item.getLocation().getFile();
+                    if (sourceFile != null)
+                    {
+                        title += " at " + sourceFile + ":" + item.getLocation().getLine();
+                        if (project != null)
+                        {
+                            foreach (string cp in project.AbsoluteClasspaths)
+                            {
+                                string pathBackSlash = cp.TrimEnd(new[]{Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar});
+                                pathBackSlash = pathBackSlash.IndexOf(Path.AltDirectorySeparatorChar.ToString()) > -1 ?
+                                    pathBackSlash + Path.AltDirectorySeparatorChar : pathBackSlash + Path.DirectorySeparatorChar;
+                                if (sourceFile.getFullPath().ToString().StartsWith(pathBackSlash))
```

Comment:
That's a bit subjective, since packages and paths should be case sensitive I'd say it shouldn't, also since we want to check for our own files, there shouldn't be casing problems. I guess that if an user adds some source path, and then renames it, the problem could pop up, but then the project would violate case sensitivity.

I know casing doesn't matter in Windows, but does in other platforms, so I like to take it into account.

@Meychi?